core/primitives/inherent_data.hpp

```diff
@@ -47,8 +47,9 @@ namespace kagome::primitives {
      */
     template <typename T>
     outcome::result<void> putData(InherentIdentifier identifier, const T& inherent) {
-      if (data.find(identifier) == data.end()) {
-        data[identifier] = common::Buffer(scale::encode(inherent).value());
+      if (data.try_emplace(identifier,
+                           common::Buffer(scale::encode(inherent).value()))
+              .second) {
```

Comment:
May keep encoding in `if`, to avoid always encoding
```suggestion
      auto [it, inserted]{data.try_emplace(identifier, common::Buffer())};
      if (inserted) {
        it->second = scale::encode(inherent).value();
```